### PR TITLE
56 add Actions/Reaction parameters in the database

### LINKED ISSUE #56: Add Action/Response parameters in the database
**Short description:**
Add the specific parameters used by actions/reactions.

**Describe the solution you'd like**
It's a JSON field which will be filled without a consitent structure.

### PR DESCRIPTION
**Pull request by ZQUEMA**
# Purpose

The purpose of this pull request is to add actions and reaction parameters in the database architecture.
Do not forget to create a migration.

Fixes #56

## Added features

* New tables are available:
* Parameter: This table stores the differents parameters without the value for the Actions and Reactions.
* ActionParameter: This table is used to store the value of the Action's parameters, and this table is used in UserHasActionsReactions model.
* ReactionParameter: This table is used to store the value of the Reaction's parameters, and this table is used in UserHasActionsReactions model.

## Updated features

* UserHasActionsReactions model has now to new relations with ActionParameters and ReactionParamaters. These tables allow to store the parameters used by the actions and reaction for each A/REA wanted by the user.

## Removed features
* The model ActionTriggers does not exist anymore. It was useless because Parameter models are more permissive.

### CODE REVIEW

File: server/prisma/schema.prisma [hunk 1]
```diff
@@ -57,25 +57,51 @@ model Reaction {
...
+  id                String              @id @unique @default(cuid())
+  name              String
+  displayName       String
+  description       String?
+  action            Action?             @relation(fields: [actionId], references: [id])
+  actionId          String?
+  reaction          Reaction?           @relation(fields: [reactionId], references: [id])
+  reactionId        String?
+  ActionParameters  ActionParameter[]
+  ReactionParameter ReactionParameter[]
```
Comment (tbellicha) [inline server/prisma/schema.prisma]: `ReactionParameter` to `ReactionParameters`

File: server/prisma/schema.prisma [hunk 2]
```diff
@@ -57,25 +57,51 @@ model Reaction {
...
+  id                String              @id @unique @default(cuid())
+  User              User                @relation(fields: [userId], references: [id])
+  Action            Action              @relation(fields: [actionId], references: [id])
+  Reaction          Reaction            @relation(fields: [reactionId], references: [id])
+  isEnable          Boolean             @default(true)
+  createdAt         DateTime            @default(now())
+  userId            String
+  actionId          String
+  reactionId        String
+  ActionParameter   ActionParameter[]
```
Comment (tbellicha) [inline server/prisma/schema.prisma]: `ActionParameter` to `ActionParameters`

File: server/prisma/schema.prisma [hunk 3]
```diff
@@ -57,25 +57,51 @@ model Reaction {
...
+  ReactionParameter ReactionParameter[]
```
Comment (tbellicha) [inline server/prisma/schema.prisma]: `ReactionParameter` to `ReactionParameters`

**Review by tbellicha — CHANGES_REQUESTED** [review]

**Review by tbellicha — APPROVED** [review]

File: server/prisma/schema.prisma [hunk 4]
```diff
@@ -7,7 +7,7 @@ generator client {
 
 generator docs {
   provider = "node node_modules/prisma-docs-generator"
-  output = "../docs/prisma"
+  output   = "../docs/prisma"
```
Comment (LucasTesnier) [inline server/prisma/schema.prisma:10]: Trailing space

**Review by LucasTesnier — CHANGES_REQUESTED** [review]
Just a typo lol

**Review by tbellicha — APPROVED** [review]

File: server/prisma/schema.prisma (hunk 4 above)
Comment (ZQUEMA) [reply server/prisma/schema.prisma:10]: That's not a trailling space 
It's the code format of prisma. The second column should be with the same indentation.

**Review by TomDUVAL-MAHE — APPROVED** [review]
NTR

**Review by LucasTesnier — APPROVED** [review]

**Pull request merged**